dotnet-desktop-guide/net/winforms/visualstudio/troubleshoot-32bit.md

```diff
@@ -0,0 +1,77 @@
+
+To summarize, 32-bit components can't be used by the Windows Forms designer in Visual Studio 2022, which is a 64-bit app. The **out-of-process designer** was created to help Windows Forms for .NET apps during design-time, and is now improved to help with loading 32-bit components.
+
+## What can you do
+
+There are a few design changes you should consider, which might help your project.
+
+- Upgrade from .NET Framework to .NET 8+.
+
+  .NET uses the out-of-process designer, which helps with 32-bit designer problems.
+
+- With .NET Framework, set your app to target `AnyCPU`.
+
+  If you target `AnyCPU` and enable `Prefer 32-bit`, your app runs under 64-bit when in Visual Studio design-time, but compiles to 32-bit for run-time.
+
+- Recompile the 32-bit component for `AnyCPU` or 64-bit.
+
+  If you have access to the source code for the 32-bit component, try compiling it for `AnyCPU` or 64-bit and reference that new version.
+
+- Find a 64-bit alternative component.
+
+  If you're using a component owned by someone else, check to see if they offer a 64-bit version, and reference that.
+
+- Try the out-of-process designer.
+
+  Your final option would be to enable the out-of-process designer for .NET Framework.
+
+## Out-of-process designer
+
+If your project targets .NET 8, you're already using the out-of-process designer. However, if you're still using .NET Framework, you need to enable the out-of-process designer.
+
+> [!WARNING]
+> The updated out-of-process 32-Bit .NET Framework Designer doesn't achieve full parity with the old in-process .NET Framework Designer due to the same architectural differences. Highly customized control designers aren't compatible. If you use custom control libraries from 3rd parties, check if they offer versions that support the out-of-process .NET Framework Designer.
+
+The **out-of-process designer** has been improved, with some limitations, to handle 32-bit issues with Visual Studio 2022:
+
+- .NET Framework benefits from improved type resolution.
+- ActiveX and COM references are supported in both .NET Framework and .NET.
+- The in-process designer in Visual Studio detects 32-bit assembly load failures and can suggest enabling the out-of-process designer.
+
+### Use the out-of-process designer
+
+Support for 32-bit references requires **Visual Studio 17.9 Preview 2 or later**. Enable it by adding the following `<PropertyGroup>` setting to your project file:
+
+```xml
+<PropertyGroup>
+    <UseWinFormsOutOfProcDesigner>True</UseWinFormsOutOfProcDesigner>
+</PropertyGroup>
+```
+
+After modifying the project file, reload the project.
+
+### 32-bit issue detection
+
+Currently, when Visual Studio detects that a 32-bit reference fails to load, it prompts you to enable the Windows Forms out-of-process designer. If you agree to enable it, the project is updated for you and then reloaded.
```

Comment:
Here is the new blog link for this feature: https://devblogs.microsoft.com/visualstudio/winforms-designer-selection-for-32-bit-net-framework-projects/